src/mod_zone_difficulty_scripts.cpp

```diff
@@ -528,14 +1350,677 @@ class mod_zone_difficulty_worldscript : public WorldScript
+            sZoneDifficulty->HardmodeInstanceData[instanceId].HardmodeOn = false;
+            sZoneDifficulty->SaveHardmodeInstanceData(instanceId);
+            sZoneDifficulty->SendWhisperToRaid("We're switching to the cinematic version of the history lesson now. (Normal mode)", creature, player);
+            CloseGossipMenuFor(player);
+        }
+
+        return true;
+    }
+
+    bool OnGossipHello(Player* player, Creature* creature) override
+    {
+        //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: OnGossipHelloChromie");
+        uint32 npcText = NPC_TEXT_OTHER;
+        if (Group* group = player->GetGroup())
+        {
+            //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: OnGossipHello Has Group");
+            if (group->IsLeader(player->GetGUID()))
+            {
+                //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: OnGossipHello Is Leader");
+                AddGossipItemFor(player, GOSSIP_ICON_CHAT, "Please Chromie, let us re-experience how all the things really happened back then. (Hard mode)", GOSSIP_SENDER_MAIN, 100);
+                AddGossipItemFor(player, GOSSIP_ICON_CHAT, "I think we will be fine with the cinematic version from here. (Normal mode)", GOSSIP_SENDER_MAIN, 101);
+
+                if (sZoneDifficulty->HardmodeInstanceData[player->GetMap()->GetInstanceId()].HardmodeOn)
+                {
+                    npcText = NPC_TEXT_LEADER_HARD;
+                }
+                else
+                {
+                    npcText = NPC_TEXT_LEADER_NORMAL;
+                }
+            }
+            else
+            {
+                creature->Whisper("I will let the leader of your group decide about this subject. You will receive a notification, when they make a new request to me.", LANG_UNIVERSAL, player);
+            }
+        }
+        SendGossipMenuFor(player, npcText, creature);
+        return true;
+    }
+};
+
+class mod_zone_difficulty_allcreaturescript : public AllCreatureScript
+{
+public:
+    mod_zone_difficulty_allcreaturescript() : AllCreatureScript("mod_zone_difficulty_allcreaturescript") { }
+
+    void OnAllCreatureUpdate(Creature* creature, uint32 /*diff*/) override
+    {
+        // Heavily inspired by https://github.com/azerothcore/mod-autobalance/blob/1d82080237e62376b9a030502264c90b5b8f272b/src/AutoBalance.cpp
+        if (!creature || !creature->GetMap())
+        {
+            return;
+        }
+
+        if ((creature->IsHunterPet() || creature->IsPet() || creature->IsSummon()) && creature->IsControlledByPlayer())
+        {
+            return;
+        }
+
+        uint32 mapId = creature->GetMapId();
+        if (sZoneDifficulty->NerfInfo.find(mapId) == sZoneDifficulty->NerfInfo.end())
+        {
+            return;
+        }
+
+        if (!creature->IsAlive())
+        {
+            return;
+        }
+
+        Map* map = creature->GetMap();
+        if (!map->IsRaid() &&
+            (!(map->IsHeroic() && map->IsDungeon())))
```

Comment:
Just gonna check. Is this script supposed to be able to be used in Normal dungeons and open world?

Reply:
No, it's hard-coded to heroic 5 man dungeons or raid dungeons.

Reply:
Oh right, I totally read the if statement wrong xD